src/libraries/System.Security.Cryptography.X509Certificates/tests/X509Certificate2PemTests.cs

```diff
@@ -0,0 +1,429 @@
+                AssertKeysMatch(TestData.RsaPkcs8Key, cert.GetRSAPrivateKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateFromPem_Rsa_Aggregate_Pkcs1_Success()
+        {
+            string pemAggregate = TestData.RsaCertificate + TestData.RsaPkcs1Key;
+            using (X509Certificate2 cert = X509Certificate2.CreateFromPem(pemAggregate, pemAggregate))
+            {
+                Assert.Equal("A33348E44A047A121F44E810E888899781E1FF19", cert.Thumbprint);
+                AssertKeysMatch(TestData.RsaPkcs1Key, cert.GetRSAPrivateKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateFromPem_Rsa_LoadsFirstCertificate_Success()
+        {
+            string certAggregate = TestData.RsaCertificate + TestData.ECDsaCertificate;
+            using (X509Certificate2 cert = X509Certificate2.CreateFromPem(certAggregate, TestData.RsaPkcs1Key))
+            {
+                Assert.Equal("A33348E44A047A121F44E810E888899781E1FF19", cert.Thumbprint);
+                AssertKeysMatch(TestData.RsaPkcs1Key, cert.GetRSAPrivateKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateFromPem_Rsa_IgnoresNonMatchingAlgorithmKeys_Success()
+        {
+            string keyAggregate = TestData.ECDsaECPrivateKey + TestData.RsaPkcs1Key;
+            using (X509Certificate2 cert = X509Certificate2.CreateFromPem(TestData.RsaCertificate, keyAggregate))
+            {
+                Assert.Equal("A33348E44A047A121F44E810E888899781E1FF19", cert.Thumbprint);
+                AssertKeysMatch(TestData.RsaPkcs1Key, cert.GetRSAPrivateKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateFromPem_Rsa_IgnoresPkcs1PublicKey_Success()
+        {
+            string keyAggregate = TestData.RsaPkcs1PublicKey + TestData.RsaPkcs1Key;
+            using (X509Certificate2 cert = X509Certificate2.CreateFromPem(TestData.RsaCertificate, keyAggregate))
+            {
+                Assert.Equal("A33348E44A047A121F44E810E888899781E1FF19", cert.Thumbprint);
+                AssertKeysMatch(TestData.RsaPkcs1Key, cert.GetRSAPrivateKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateFromPem_Rsa_IgnoresPkcs8PublicKey_Success()
+        {
+            string keyAggregate = TestData.RsaPkcs8PublicKey + TestData.RsaPkcs1Key;
+            using (X509Certificate2 cert = X509Certificate2.CreateFromPem(TestData.RsaCertificate, keyAggregate))
+            {
+                Assert.Equal("A33348E44A047A121F44E810E888899781E1FF19", cert.Thumbprint);
+                AssertKeysMatch(TestData.RsaPkcs1Key, cert.GetRSAPrivateKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateFromPem_Rsa_KeyMismatch_Fail()
+        {
+            AssertExtensions.Throws<ArgumentException>("privateKey", () =>
+                X509Certificate2.CreateFromPem(TestData.RsaCertificate, TestData.OtherRsaPkcs1Key));
```

Comment:
While it makes sense for CopyWithPrivateKey to call the mismatch an ArgumentException (particularly since it already shipped), "privateKey" wasn't an argument here, and the implementation should probably turn the called ArgumentExceptions into CryptographicExceptions.

Reply:
If only someone recently co-authored a book about designing a framework, I could learn all of this in advance.

Will fix.